### PR TITLE
feat(optimize): add MPS peak recovery metrics

### PR DESCRIPTION
**Pull request by enarjord**
## Summary

- add Apple MPS proxy scoring for `peak_recovery_hours_pnl`, `peak_recovery_days_pnl`, and `peak_recovery_days_strategy_eq`
- track strict cumulative net-PnL peak recovery at every Metal proxy fill, including the final analyzed tail
- support single-coin and one-sided multi-coin EMA Anchor and Trailing Martingale runs while failing closed for dual-side multi-coin metrics that cannot be reconstructed from independent directional dispatches
- keep weighted daily fill volume unsupported because the compact daily proxy surface cannot safely reconstruct fill-filtered intraday suffix boundaries
- keep exact Rust validation authoritative and document the added surface and topology boundary

## Validation

- 808 non-MPS optimizer tests passed after the review fix
- 212 physical Apple MPS tests passed across single-coin long, short, and dual-side plus one-sided multi-coin EMA Anchor and Trailing Martingale paths
- 262 Rust tests passed; `cargo fmt --check` and default/no-default `cargo check` passed
- bounded cached two-coin end-to-end MPS optimizations completed for both strategy kinds: 1,024 proxy evaluations, 64 exact validations, eight generations, no drift, constraint-classification, or shutdown halt
- documentation validation completed with zero errors (two pre-existing repository budget warnings)
- rebuilt extension source fingerprint and loaded artifact stamp both verified as `629d241ddba2dbb35751053a318c0bbe55f070131cccaf8934e664bf028b471d`

### CODE REVIEW

**Comment by enarjord** [comment]
@codex review

File: src/optimization/gpu/metrics.py [hunk 1]
```diff
@@ -874,6 +878,43 @@ def _weighted_pnl_metrics(
...
+    eligible &= fill_count > 1.0
+    total = torch.zeros(
+        day_volume.shape[0], dtype=day_volume.dtype, device=day_volume.device
+    )
+    for subset in subsets:
+        fill_days = subset & day_has_fill
+        count = fill_days.sum(dim=1)
+        value = torch.where(
+            fill_days, day_volume, torch.zeros_like(day_volume)
+        ).sum(dim=1) / count.clamp(min=1).to(day_volume.dtype)
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/gpu/metrics.py]: Confirmed. The compact daily surface cannot reconstruct the exact intraday suffix boundary safely. Commit 5fb7e46ab removes volume_pct_per_day_avg_w from the MPS supported surface, docs, and tests, so GPU optimization now fails closed for that metric. The broad optimizer suite passes after the change.

**Comment by enarjord** [comment]
@codex review
